server/controllers/profilephoto.js

```diff
@@ -0,0 +1,99 @@
+const S3 = require("aws-sdk/clients/s3");
+const fs = require("fs");
+const Profile = require("../Models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route Post /photourl
+// @desc Save user photo to AWS S3 bucket and remove the previous one to save storage
+// @access Private
+exports.uploadPhoto = asyncHandler(async (req, res) => {
+    const userId = req.user.id;
+
+    const profile = await Profile.findOne({ userId });
+    if (!profile) {
+        res.status(404);
+        throw new Error("Please create your profile first");
+    }
+    const s3 = new S3({
+        region: process.env.AWS_REGION,
+        accessKeyId: process.env.AWS__ACCESS_KEY,
+        secretAccessKey: process.env.AWS_SECRET_KEY,
+    })

```

Comment:
This seems better placed outside of this function, even better to initialize it once only once (e.g. initialize in app).